cli/command/system/prune.go

```diff
@@ -119,12 +127,23 @@ func confirmationMessage(options pruneOptions) string {
 			warnings = append(warnings, "all dangling build cache")
 		}
 	}
-	if len(options.filter.String()) > 0 {
-		warnings = append(warnings, "Elements to be pruned will be filtered with:")
-		warnings = append(warnings, "label="+options.filter.String())
+
+	var filters []string
+	pruneFilters := command.PruneFilters(dockerCli, options.filter.Value())
+	if pruneFilters.Len() > 0 {
+		// TODO remove fixed list of filters, and print all filters instead,
+		// because the list of filters that is supported by the engine may evolve over time.
+		for _, name := range []string{"label", "label!", "until"} {
+			for _, v := range pruneFilters.Get(name) {
+				filters = append(filters, name+"="+v)
+			}
+		}
+		sort.Slice(filters, func(i, j int) bool {
```

Comment:
Prevent the test from being flaky, as the filters are stored as a map, so order is randomised